packages/forms/backend/sirius-components-collaborative-forms/src/main/java/org/eclipse/sirius/components/collaborative/forms/configuration/FormEventProcessorFactoryConfiguration.java

```diff
@@ -0,0 +1,65 @@
+/*******************************************************************************
+ * Copyright (c) 2023 Obeo.
+ * This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v2.0
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *     Obeo - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.sirius.components.collaborative.forms.configuration;
+
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+
+import org.eclipse.sirius.components.collaborative.api.IRepresentationEventProcessorFactory;
+import org.eclipse.sirius.components.collaborative.forms.api.IFormEventHandler;
+import org.eclipse.sirius.components.collaborative.forms.api.IFormProcessor;
+import org.eclipse.sirius.components.collaborative.forms.api.IWidgetSubscriptionManagerFactory;
+import org.eclipse.sirius.components.core.api.IObjectService;
+import org.springframework.context.annotation.Configuration;
+
+/**
+ * Bundles the common dependencies that most {@link IRepresentationEventProcessorFactory} implementations for Form Representation need
+ * into a single object for convenience.
+ *
+ * @author frouene
+ */
+@Configuration
+public class FormEventProcessorFactoryConfiguration {
+
+    private final IObjectService objectService;
+
+    private final List<IFormEventHandler> formEventHandlers;
+
+    private final IWidgetSubscriptionManagerFactory widgetSubscriptionManagerFactory;
+
+    private final Optional<IFormProcessor> optionalFormProcessor;
+
+    public FormEventProcessorFactoryConfiguration(IObjectService objectService, List<IFormEventHandler> formEventHandlers, IWidgetSubscriptionManagerFactory widgetSubscriptionManagerFactory, Optional<IFormProcessor> optionalFormProcessor) {
+        this.objectService = Objects.requireNonNull(objectService);
+        this.formEventHandlers = Objects.requireNonNull(formEventHandlers);
+        this.widgetSubscriptionManagerFactory = Objects.requireNonNull(widgetSubscriptionManagerFactory);
+        this.optionalFormProcessor = Objects.requireNonNull(optionalFormProcessor);
+    }
+
+    public IObjectService getObjectService() {
+        return this.objectService;
+    }
+
+    public List<IFormEventHandler> getFormEventHandlers() {
+        return this.formEventHandlers;
+    }
+
+    public IWidgetSubscriptionManagerFactory getWidgetSubscriptionManagerFactory() {
+        return this.widgetSubscriptionManagerFactory;
+    }
+
+    public Optional<IFormProcessor> getOptionalFormProcessor() {
+        return this.optionalFormProcessor;
+    }
```

Comment:
```java
public IFormProcessor getFormProcessor() {
        return this.optionalFormProcessor.orElse(new IFormProcessor.NoOp());
}
```
would avoid dealing with optionals and nullable fields in the rest of the code.

Reply:
Done